ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/SandboxServer.scala

```diff
@@ -157,23 +153,27 @@ object SandboxServer {
         .migrate()
     }
 
-  // TODO SoX-to-sandbox-classic: Work-around to emulate the ledgerIdMode used in sandbox-classic.
-  //                              This is needed for the Dynamic ledger id mode, when the index should be initialized with the existing ledger id (only used in testing).
-  private def getLedgerId(
-      jdbcUrl: String
-  )(implicit
-      resourceContext: ResourceContext,
-      executionContext: ExecutionContext,
-      materializer: Materializer,
-  ): resources.Resource[Option[String]] =
+  // Work-around to emulate the ledgerIdMode used in sandbox-classic.
+  // This is needed for the Dynamic ledger id mode, when the index should be initialized with the existing ledger id (only used in testing).
+  private def getLedgerId(jdbcUrl: String): Option[String] =
     newLoggingContext { implicit loggingContext: LoggingContext =>
-      Resource
-        // TODO Sandbox: Handle verbose error logging on non-existing ledger id (e.g. when starting on non-initialized Index DB)
-        .fromFuture(IndexMetadata.read(jdbcUrl, ErrorFactories(true)).transform {
-          case Failure(_) => Success(None)
-          case Success(indexMetadata) =>
-            if (indexMetadata.ledgerId.isEmpty) Success(None)
-            else Success(Some(indexMetadata.ledgerId))
-        })
+      Try {
+        val dbType = DbType.jdbcType(jdbcUrl)
+        val storageBackendFactory = StorageBackendFactory.of(dbType)
+        val dataSource =
+          storageBackendFactory.createDataSourceStorageBackend.createDataSource(jdbcUrl)
+
+        storageBackendFactory.createParameterStorageBackend
+          .ledgerIdentity(dataSource.getConnection)
+          .map(_.ledgerId)
+      } match {
+        case Failure(err) =>
+          logger.warn(
+            s"Failure encountered trying to retrieve ledger id: ${err.getMessage}. Assuming uninitialized index."
+          )
+          None
+        case Success(maybeLedgerId) =>
+          maybeLedgerId.map(Tag.unwrap).filter(_.nonEmpty)
+      }
```

Comment:
When initializing on an new DB, using `IndexMetadata.read(..)` throws and logs an ERROR with `Table 'parameters' not found..` In order to avoid the confusing logging on this frequently encountered use-case, I've bootstrapped the storage backend and the data-source directly. Then, I log minimally the error message.